install.sh

```diff
@@ -38,63 +61,177 @@ while [ "$1" != "" ]; do
+    exit_if_user_declines 
+    "$BREW_COMMAND" install pkg-config
   fi
-  "$BREW_COMMAND" install pkg-config coreutils gcc@$GCC_VER
+  if [ -z ${REALPATH+x} ] || [ -z ${MAKE+x} ] ; then
+    ask_homebrew_package_permission "realpath and make" "coreutils"
+    exit_if_user_declines 
+    "$BREW_COMMAND" install coreutils
+  fi
+
+  CC=`which gcc-$GCC_VER`
+  CXX=`which g++-$GCC_VER`
+  FC=`which gfortran-$GCC_VER`
 fi
-export CC=`which gcc-$GCC_VER`
-export CXX=`which g++-$GCC_VER`
-export FC=`which gfortran-$GCC_VER`
 
 PREFIX=`realpath $PREFIX`
 
-GASNET_TAR_FILE="GASNet-stable.tar.gz"
-GASNET_SOURCE_URL="https://bitbucket.org/berkeleylab/gasnet/downloads/$GASNET_TAR_FILE"
-DEPENDENCIES_DIR="build/dependencies"
-if [ ! -d $DEPENDENCIES_DIR ]; then
-  mkdir -pv $DEPENDENCIES_DIR
+export FPM_FC="$FC"
+export FPM_CC="$CC"
+FPM_SOURCE_URL="https://github.com/fortran-lang/fpm/archive/refs/tags/v$FPM_VERSION.tar.gz"
+
+if [ ! -d build/dependencies ]; then
+  mkdir -p build/dependencies
 fi
 
-cd $DEPENDENCIES_DIR
+ask_package_permission()
+{
+  echo ""
+  echo "$1 not found."
+  echo ""
+  echo "Press 'Enter' for the square-bracketed default answer:"
+  printf "Is it ok to download and install $1? [yes] "
+}
 
-  if [ ! -f $GASNET_TAR_FILE ]; then
-    curl -L $GASNET_SOURCE_URL > $GASNET_TAR_FILE
-  fi
+if command -v fpm > /dev/null 2>&1; then
+  FPM="fpm"
+else
+  ask_package_permission "fpm"
+  exit_if_user_declines
+  curl -L $FPM_SOURCE_URL | tar xvz -C build/dependencies/
+  (cd build/dependencies/fpm-$FPM_VERSION && ./install.sh --prefix="$PREFIX")
+fi
 
-  if [ ! -d GASNet-stable ]; then
-    tar xf $GASNET_TAR_FILE
+export PKG_CONFIG_PATH="$PREFIX"/lib/pkgconfig
+pkg="gasnet-smp-seq"
+if [ ! -f "$PKG_CONFIG_PATH/$pkg.pc" ]; then
+  echo ""
+  echo "GASNet-EX $pkg.pc file not found in $PKG_CONFIG_PATH."
+  echo ""
+  echo "Press 'Enter' for the square-bracketed default answer:"
+  printf "Is it ok to download and install GASNet-EX? [yes] "
+  read answer
+  if [ -n "$answer" -a "$answer" != "y" -a "$answer" != "Y" -a "$answer" != "Yes" -a "$answer" != "YES" -a "$answer" != "yes" ]; then
+    echo "Installation declined."
+    echo "Please ensure the $pkg.pc file is in $PKG_CONFIG_PATH and then rerun './install.sh'."
+    echo "Caffeine was not installed."
   fi
 
-  if [ ! -d gasnet ]; then
-    mkdir -v gasnet
-    cd gasnet
-      ../GASNet-stable/configure --prefix "$PREFIX"
+  GASNET_TAR_FILE="GASNet-$GASNET_VERSION.tar.gz"
+  GASNET_SOURCE_URL="https://gasnet.lbl.gov/EX/GASNet-$GASNET_VERSION.tar.gz"
+  DEPENDENCIES_DIR="build/dependencies"
+  if [ ! -d $DEPENDENCIES_DIR ]; then
+    mkdir -pv $DEPENDENCIES_DIR
+  fi
+  
+  curl -L $GASNET_SOURCE_URL | tar xvz - -C $DEPENDENCIES_DIR
+  
+  if [ -d $DEPENDENCIES_DIR/GASNet-$GASNET_VERSION ]; then
+    cd $DEPENDENCIES_DIR/GASNet-$GASNET_VERSION
+      FC="$FC" CC="$CC" CXX="$CXX" ../GASNet-$GASNET_VERSION/configure --prefix "$PREFIX"
       make -j 8 all
       make check
```

Comment:
GASNet `make check` builds a bunch of GASNet tests (which ensures they link), but (with a few trivial exceptions) does not run them.  Running multi-node GASNet tests often requires a working spawning setup or even a batch environment, so separate command `make run-tests` is used to actually run the tests.

The GASNet test building step takes a significant amount of disk space and time (especially with serial make), and running the tests would take even longer. This might make sense inside a caffeine self-test/CI, but really doesn't make sense for an installer. Given the caffeine build system just wants to use GASNet (not test it), I'd suggest deleting this `make check` step. 

On a related note, the `make install` step below can also benefit from parallel make eg `-j 8`.

Reply:
`make check` has been dropped in the latest commit.